sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs

```diff
@@ -54,25 +70,36 @@ public static CommunicationIdentifier FromRawId(string rawId)
         {
             Argument.AssertNotNullOrEmpty(rawId, nameof(rawId));
 
-            if (rawId.StartsWith("4:", StringComparison.OrdinalIgnoreCase))
+            if (rawId.StartsWith(Phone, StringComparison.OrdinalIgnoreCase))
             {
-                return new PhoneNumberIdentifier(rawId.Substring("4:".Length));
+                return new PhoneNumberIdentifier(rawId.Substring(Phone.Length));
             }
 
             var segments = rawId.Split(':');
             if (segments.Length < 3)
+            {
+                if (segments.Length == 2 && segments[0] == "28")
+                {
+                    return new MicrosoftBotIdentifier(segments[1], true, CommunicationCloudEnvironment.Public);
+                }
                 return new UnknownIdentifier(rawId);
+            }
 
             var prefix = $"{segments[0]}:{segments[1]}:";
-            var suffix = rawId.Substring(prefix.Length);
+            var suffix = segments[2];
 
             return prefix switch
             {
-                "8:teamsvisitor:" => new MicrosoftTeamsUserIdentifier(suffix, true),
-                "8:orgid:" => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Public),
-                "8:dod:" => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Dod),
-                "8:gcch:" => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Gcch),
-                "8:acs:" or "8:spool:" or "8:dod-acs:" or "8:gcch-acs:" => new CommunicationUserIdentifier(rawId),
+                TeamUserAnonymous => new MicrosoftTeamsUserIdentifier(suffix, true),
+                TeamUserPublicCloud => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Public),
+                TeamUserDodCloud => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Dod),
+                TeamUserGcchCloud => new MicrosoftTeamsUserIdentifier(suffix, false, CommunicationCloudEnvironment.Gcch),
+                AcsUser or SpoolUser or AcsUserDodCloud or AcsUserGcchCloud => new CommunicationUserIdentifier(rawId),
+                BotGcchCloudGlobal => new MicrosoftBotIdentifier(suffix, true, CommunicationCloudEnvironment.Gcch),
+                BotPublicCloud => new MicrosoftBotIdentifier(suffix, false, CommunicationCloudEnvironment.Public),
+                BotDodCloudGlobal => new MicrosoftBotIdentifier(suffix, true, CommunicationCloudEnvironment.Dod),
+                BotGcchCloud => new MicrosoftBotIdentifier(suffix, false, CommunicationCloudEnvironment.Gcch),
+                BotDodCloud => new MicrosoftBotIdentifier(suffix, false, CommunicationCloudEnvironment.Dod),
```

Comment:
I would expect the constants to be replaced in the [`MicrosoftTeamsUserIdentifier`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.Common/src/MicrosoftTeamsUserIdentifier.cs) and other identifier classes as well.

Reply:
OK